lightning/restore/restore.go

```diff
@@ -793,11 +804,47 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	wg.Wait()
 	close(stopPeriodicActions)
 
-	err := restoreErr.Get()
+	err = restoreErr.Get()
 	logTask.End(zap.ErrorLevel, err)
 	return err
 }
 
+func (rc *RestoreController) newChecksumManager() (ChecksumManager, error) {
+	pdAddr := rc.cfg.TiDB.PdAddr
+	pdVersion, err := common.FetchPDVersion(rc.tls, pdAddr)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// for v4.0.0 or upper, we can use the gc ttl api
+	var manager ChecksumManager
+	if pdVersion.Major >= 4 {
```

Comment:
I remember it's not all 4.0 version support gc-ttl...

Reply:
since `v4.0.0.rc.1` pd has supported this interface, so I think this check is ok?